src/libsyncengine/jobs/abstractjob.cpp

```diff
@@ -57,6 +57,19 @@ ExitCode AbstractJob::runSynchronously() {
     return _exitCode;
 }
 
+void AbstractJob::setProgress(int64_t newProgress) {
+    _progress = newProgress;
+    if (_progressPercentCallback) {
+        if (_expectedFinishProgress == -1) {
```

Comment:
How often is this warning going to be issued? Would it be wiser to write it only in the extended logs?

Reply:
What about 
- throwing an exception instead
- writing an assertion failure
- writing only on log entry for the same problem (if possible) ?

Reply:
To preserve the abstraction principle, I prefer the last option. I also set a default behaviour of returning 100% in case we could not calculate the progress. You can take a look at the change here: [07dcbd3](https://github.com/Infomaniak/desktop-kDrive/pull/139/commits/07dcbd3a5fe603e5954e4543b1965aeb95d84490)